src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -223,6 +224,20 @@ private bool MatchString(string str)
             {
```

Comment:
Not you, but `c` is a confusing name for an offset. It should probably be something like `i` or `pos`. 2 places in this file.

Reply:
Also incidentally confusing: `Notoneloop` should really be `Anyloop` (and same for `Notonerep` and `Notonelazy` but not `Notone`, which really is not..)

Reply:
Oh for sure. There are so many terribly named variables in this library